docs/runbooks/monthly-release-to-editors-and-webmasters.md

```diff
@@ -20,36 +20,56 @@ release - 1. This usually mean last weeks release.
-6. Start `dplsh` from the `/infrastructure` directory of your local
-   environment by running `../tools/dplsh/dplsh.sh`
-7. Run `task cluster:mode:release` to get more nodes while deploying.
-8. Run `task sites:sync` from `dplsh` to deploy the changes.
-9. If there are any Terraform changes then do not apply them, abort the
-   deployment and consult the platform team.
-10. Open the Deployments list page within the Lagoon UI to see all running and
-   queued deployments.
-11. Wait for all the deployments to complete.
-12. Run `task sites:incomplete-deployments` to identify any failed deployments.
-13. If some deployments did not complete determine if the error relates to the
-    platform or the application.
-14. For all platform-related errors try to redeploy the environment from
-    the Lagoon UI.
-15. Merge the pull request once the deployment completes.
-16. Run `task cluster:adjust:resource-request` from `dplsh`.
-17. Run `task cluster:promote-workloads-to-prod` from `dplsh`.
-18. Reset the webmaster moduletest environments.
-19. Run 'task cluster:mode:reset' from `dplsh`.
-
-## Procedure: a some sites fails to deploy
+3. Release to webmaster production:
+    1. Update 'dpl-cms-release' of the 'x-webmaster' anchor to the penultimate
+        version of the CMS.
+    2. Commit the change and push your branch to GitHub and create a pull request.
+    3. Request a review for the change and wait for approval.
+    4. Start `dplsh` from the `/infrastructure` directory of your local
+       environment by running `../tools/dplsh/dplsh.sh`
+    5. Run `task cluster:mode:release` to get more nodes while deploying.
+    6. Run `task sites:sync` from `dplsh` to deploy the changes.
+    7. If there are any Terraform changes then do not apply them, abort the
+       deployment and consult the platform team.
+    8. Open the Deployments list page within the Lagoon UI to see all running and
+       queued deployments.
+    9. Wait for all the deployments to complete.
+    10. Run `task sites:incomplete-deployments` to identify any failed deployments.
+    11. If some deployments did not complete determine if the error relates to the
+        platform or the application.
+    12. For all platform-related errors try to redeploy the environment from
+        the Lagoon UI.
+    13. Merge the pull request once the deployment completes.
+    14. Synchronize moduletest with main,so moduletest becomes an exact
+        copy of main by running `task sites:webmaster:reset-moduletest`.
+4. Create a new branch from `main`.
+5. Release to editors and webmaster moduletest:
+    1. Now update `infrastructure/environments/dplplat01/sites.yaml`:
+    2. The 'x-defaults' anchors' 'dpl-cms-release' tag should be bumped to the
+        new version.
+    3. Then update the 'moduletest-dpl-cms-release' of 'x-webmaster'
+        and 'x-webmaster-with-defaults' anchor with the newest release.
+    4. Commit the change and push your branch to GitHub and create a pull request.
```

Comment:
I do not understand this structure.

As I see it we previously deployed the right version to all environments (main for editors, main and moduletest for webmasters) in one go, one branch, one run of each support task (`cluster:mode`, `cluster:adjust` etc.) one `task sites:sync` etc.

The current suggestion implies that we deploy with some elements done once (support tasks) and other elements (branch, `task sites:sync` etc.) done twice. I do not understand why this is needed.

To me there are challenges with the suggestion:

1. Monthly deployments now entail many more manual steps. This is more complex, error prone and time consuming.
2. There is more duplication between the steps in the weekly and monthly deployment which may come out of sync with future changes.

Bear in mind that it is some weeks ago since I deployed myself so I do not know if there are any other reasons to run the process as suggested. The PR mentions a right sequence but I do understand what is wrong with the current suggestion :smile:.

Reply:
> when we do the synchronization between the databases the sites must be on the same version

That is not entirely correct. The database must the the same version *or older*. If we sync the database from a site running an older version of the project to a site running a newer version and run a deployment on the site then database will be updated to the new version.

So: If we sync a production site running 43.1 into a moduletest running 44.0 then we must run a deployment afterwards to migrate it from 43.1 to 44.0.

This should be included in the `sites:webmaster:reset-moduletest` task.

Reply:
With this in mind with should also update https://github.com/danskernesdigitalebibliotek/dpl-platform/blob/557f10b6c6d2e7e61f9e0458b16a7acf2f2a37fd/docs/runbooks/ui-sync-site-state.md#L61-L61 accordingly

Reply:
for the sake of my understanding, what would happen if we synced moduletest v. 44.0 to main v.43.1 and then deployed main? Mikkel was very specific about versions having to be identical to minimize the chance of having issues